vitamins/bearing_block.scad

```diff
@@ -169,3 +170,147 @@ module scs_bearing_block_assembly(type, part_thickness, screw_type, nut_type) {
+                        translate([-(W/2) + chamfer, -((H-h))]) {
+                            difference() {
+                                union() {
+                                    right_trapezoid(Ah - chamfer,Ah,chamfer);
+                                    rotate([0,0,90])
+                                        right_trapezoid(A + chamfer,A,chamfer);
+                                }
+                                translate([-(chamfer +0.1), -0.1,0])
+                                    right_trapezoid(chamfer/2 + 0.2, 0, chamfer/2+ 0.2);
+                            }
+                        }
+                    }
+            }
+            // bolt holes
+            for(x = [-J / 2, J / 2])
+                translate([x, h - H + I/2 - 0.1, 0])
+                    rotate([90, 0, 0])
+                        for (y = [-K / 2, K / 2])
+                            translate([0, y])
+                                cylinder(r = boltHoleRadius, h=I + 0.1, center=true);
+            // set screw holes
+            translate([-width/2 + S2L/2 - 0.1,0,0])
+                rotate([0,90,0])
+                    cylinder(r = setScrewRadius, h=S2L, center=true);
+            translate([0,-(H-h) + chamfer + S2L/2 -0.1,0])
+                rotate([0,90,90])
+                    cylinder(r = setScrewRadius, h=S2L, center=true);
+
+        }
+    }
+    not_on_bom() no_explode() {
+        spacer = sbr_spacer(type);
+        for(end = spacer ? [-1, 1] : 0)
+            translate_z(end * (bearing_length(bearing) + spacer) / 2)
+                rotate([0,0,-90])
+                    linear_bearing(bearing);
+
+        for(end = [-1, 1])
+            translate_z(end * ((spacer ? 2 * bearing_length(bearing) + spacer : bearing_length(bearing)) + circlip_thickness(clip)) / 2)
+                rotate(180)
+                    internal_circlip(clip);
+
+        translate([-width/2 - 0.1,0,0])
+            rotate([0,90,180])
+               screw(S2, S2L);
+        translate([0,-(H-h) + chamfer - 0.1,0])
+            rotate([0,90,270])
+                screw(S2, S2L);
+    }
+}
+
+module sbr_bearing_block_hole_positions(type) { //! Place children at hole positions
+    //y = sbr_size(type).y;
+    h = sbr_hole_offset(type);
+    H = sbr_side_height(type);
+    bearing_block_hole_positions(type, h-H)
+        children();
+}
+
+module sbr_bearing_block_assembly(type, part_thickness, screw_type) { //! Assembly with screws in place
+
+    I = type[11]; // screw length
+    sbr_bearing_block(type);
+    screw_type = is_undef(screw_type) ? sbr_screw(type) : screw_type;
+    G = sbr_size(type)[1];
+
+    screw_length = screw_shorter_than(I + part_thickness);
+
+    translate([0,-part_thickness,0])
+        sbr_bearing_block_hole_positions(type)
+            rotate([0,180,0])
+                screw(screw_type, screw_length);
+}
+
+/* Generic bearing block functions */
+
+module bearing_block_hole_positions(type, y) { //! Place children at hole positions
```

Comment:
This doesn't seem to be generic. All the scs blocks now have their screws in the sbr positions, which is wrong.

Reply:
Weird I didn't catch that, as I did check the results on the SCS blocks... Looking into it.

Reply:
> This doesn't seem to be generic. All the scs blocks now have their screws in the sbr positions, which is wrong.

It should be fixed now. I removed the generic function as after the change the functions had very little in common now.